modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankedListQualityMetric.java

```diff
@@ -38,7 +39,9 @@
  *
  * RelevancyLevel specifies the type of object determining the relevancy level of some known docid.
  * */
-public abstract class RankedListQualityMetric extends ToXContentToBytes implements NamedWriteable {
+public abstract class RankedListQualityMetric<T extends RankedListQualityMetric<?>>
```

Comment:
Not sure if I would add the generic type here, but I think it is needed for `FromXContentBuilder<T>`? I'm not sure how the alternatives look like.